client/asset/eth/multirpc.go

```diff
@@ -784,11 +781,26 @@ func errorFilter(err error, matches ...interface{}) bool {
 }
 
 // withOne runs the provider function against the providers in order until one
-// succeeds or all have failed.
-func (m *multiRPCClient) withOne(providers []*provider, f func(*provider) error, acceptabilityFilters ...acceptabilityFilter) (superError error) {
+// succeeds or all have failed. The context used to run functions has a time
+// limit equal to defaultRequestTimeout for all requests to return. If
+// operations are expected to run longer than that the calling function should
+// not use the altered context.
+func (m *multiRPCClient) withOne(ctx context.Context, providers []*provider, f func(context.Context, *provider) error, acceptabilityFilters ...acceptabilityFilter) (superError error) {
```

Comment:
So, I needed a ctx to check the header time... and so I thought it might be better to set the context deadline in this function to ensure all the functions that use providers have a deadline. I think it is easier to keep up with than setting them all separately. You can see a couple cases where this timeout is not used, but in the vast majority we are just running one rpc quest with these so I think this is cleaner. This isn't strictly needed to solve the issue however.

Reply:
Ok. The extra ctx in `currentFees` and `transactionConfirmations` were just because two requests were being done with the same limit. But, they probably only should take milliseconds, so probably fine. I was thinking `nextNonce` was using the same context the whole time, but now that I look again it is using a new `withPreferred` every time so a new timeout. I was just confused it seems.